packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -242,13 +250,29 @@ module.exports = (
       }
     }
 
-    async function getTableOfContents(markdownNode, pathToSlugField) {
+    async function getTableOfContents(
+      markdownNode,
+      pathToSlugField,
+      maxDepth,
+      heading,
+      tight
+    ) {
       const cachedToc = await cache.get(tableOfContentsCacheKey(markdownNode))
+      const appliedTocOptions = tocOptions
+      if (maxDepth && _.isNumber(maxDepth) && _.inRange(maxDepth, 0, 7)) {
```

Comment:
with `maxDepth` set defaultValue to 6 in graphql field definition - this will always be used over pluginOptions

Reply:
you would need to remove `defaultValue` from graphql definition or use pluginOptions to set `defaultValue` there instead of hardcoding values

Reply:
I hopefully fixed this with the new commit.